tests/e2e/e2e_setup_test.go

```diff
@@ -185,10 +191,17 @@ func (s *IntegrationTestSuite) SetupSuite() {
 		}
 	}
 
-	if !skipUpgrade {
+	switch {
```

Comment:
I don't understand the conditions of this switch.

`!skipUpgrade && !isFork` -> we are supposed to do both but, in fact, only regular upgrade is run

`skipUpgrade` fork can be either skipped or unskipped -> we only run post upgrade tests without doing the fork upgrade

What's going on here?

Reply:
Is it possible to decouple `runValidators` from any fork or upgrade logic?

From single responsibility principle, `runValidators` should be responsible for one task - running the validators. Any other setup or configuration logic should be happening separately. It feels that this coupling makes it more difficult to reason about the relationship of this configuration to how and when the upgrade and fork logic is done.